File: python_backend/game_logic/threatArea/get_king_locs.py
from game_logic.piece_type.get_piece_type import get_piece_type
from game_logic.test_objects.sample_board_dicts import sample_board_dicts
from game_logic.color.get_color import get_color


def get_king_locs(board, color):
    """get the location of the king of given color NOTE: in the modified chess can return list instead of single item"""
    locs = list(filter(lambda s: get_piece_type(board[s]) == 'K' and get_color(board[s]) == color, board.keys()))
    if len(locs) == 1:
        return locs[0]
    else:
        return locs


if __name__ == "__main__":
    board = sample_board_dicts['check_example3']
    print(get_king_locs(board, 'W') == (4, 4))
    print(get_king_locs(board, 'B') == (4, 8))

File: node.js/threatArea/getKingLocs.js
const  { rfToXy } = require("../coordType/crdCnvrt")
const  getPieceType  = require("../pieceType/getPieceType");
const mapListRfToXy = require("../coordType/mapListRfToXy");
const  getColor  = require("../color/getColor");



function getKingLocs(board, color) {
    /*get the location of the king of given color NOTE: in the modified chess can return list instead of single item**/
    var locs = Object.keys(board).filter(s => getPieceType(board[s]) === 'K' &&  getColor(board[s]) === color)
    if (locs.length === 1) {
        return rfToXy(locs[0])
    }
    else {
        return mapListRfToXy(locs)
    }
}

module.exports = getKingLocs;

if (require.main === module) {
    
    const  sampleBoardDicts  = require("../testObjects/sampleBoardDicts");
    const printBoard = require("../printers/printBoard");

    var board, kingLoc;
    
    board = sampleBoardDicts['check_example3']
    kingLoc = getKingLocs(board, 'W')
    printBoard(board, `\ncheck_example3, white`, [kingLoc])
    kingLoc = getKingLocs(board, 'B')
    printBoard(board, `\ncheck_example3, black`, [kingLoc])

    board = sampleBoardDicts['super_checkmate_impossible_example']
    kingLoc = getKingLocs(board, 'W')
    printBoard(board, `\nsuper_checkmate_impossible_example, white`, [kingLoc])
    kingLoc = getKingLocs(board, 'B')
    printBoard(board, `\nsuper_checkmate_impossible_example, black`, [kingLoc])

}
